src/infrastructure/fileUploader.repository.ts

```diff
@@ -0,0 +1,58 @@
+import type NotesApiTransport from './transport/notes-api';
+import type { Note } from '@/domain/entities/Note';
+import mime from 'mime';
+import type FileUploaderRepositoryInterface from '@/domain/fileUploader.repository.interface';
+
+/**
+ * Facade for interacting with files
+ */
+export default class FileUploaderRepository implements FileUploaderRepositoryInterface {
+  /**
+   * Transport instance
+   */
+  private readonly transport: NotesApiTransport;
+
+  /**
+   * Repository constructor
+   * @param notesApiTransport - notes api transport instance
+   */
+  constructor(notesApiTransport: NotesApiTransport) {
+    this.transport = notesApiTransport;
+  }
+
+  /**
+   * Upload file data
+   * @param noteId - note indentifier to make attachment
+   * @param fileData - file binary data
+   * @param fileName - file name
+   */
+  public async upload(noteId: Note['id'], fileData: Blob, fileName?: string): Promise<string> {
+    const form = new FormData();
+
+    /**
+     * Get file mime type and define extension by it
+     */
+    const mimeType = fileData.type;
+    const fileExtension = mime.getExtension(mimeType);
+
+    /**
+     * Define file name, if it is not passed, use note id for name
+     */
+    const name = fileName ?? noteId;
```

Comment:
noteId is not unique, maybe use `nanoId()` instead ? Or always use nano id and omit original file name

Reply:
It is not necessary to have unique file name

Reply:
so why we need a filename at all?

Reply:
it is necessary for form data field and for additional info, which stores in database

Reply:
I think, giving a nanoId() would be better

Reply:
done